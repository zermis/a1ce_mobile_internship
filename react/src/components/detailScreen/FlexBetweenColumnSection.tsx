import React from 'react';
import { StyleSheet, Text, View } from 'react-native';

import { detailPageStyle } from '../../shared/Styles';

interface Props {
    header?: string;
    body?: JSX.Element;
    style?: object;
}

const FlexBetweenColumnSection = (props: Props): JSX.Element => {
    return (
        <View style={styles.container}>
            <Text style={{ ...detailPageStyle.textHeader, ...props.style }}>
                {props.header}
            </Text>
            {props.body}
        </View>
    );
};

const styles = StyleSheet.create({
    container: {
        marginVertical: 9
    }
});

export default FlexBetweenColumnSection;
