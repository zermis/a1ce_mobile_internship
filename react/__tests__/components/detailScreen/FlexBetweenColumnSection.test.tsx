import FlexBetweenColumnSection from '../../../src/components/detailScreen/FlexBetweenColumnSection';
import renderer from 'react-test-renderer';
import { Text } from 'react-native';

it('render flex between column section correctly', () => {
    const tree = renderer
        .create(
            <FlexBetweenColumnSection
                header="Header"
                body={<Text>Body</Text>}
            />
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});
