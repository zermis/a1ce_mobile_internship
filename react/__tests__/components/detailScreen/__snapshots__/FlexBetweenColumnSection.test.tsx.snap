// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render flex between column section correctly 1`] = `
<View
  style={
    Object {
      "marginVertical": 9,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#353334",
        "fontFamily": "ibm-semibold",
        "fontSize": 16,
        "marginBottom": 9,
      }
    }
  >
    Header
  </Text>
  <Text>
    Body
  </Text>
</View>
`;
